lib/screens/home/home_bloc.dart

```diff
@@ -69,6 +77,12 @@ class HomeBloc extends Bloc<HomeEvent, HomeState> {
     }
     return HomePreferences(sortType: sortType, filterType: filterType);
   }
+
+  @override
+  HomeState fromJson(Map<String, dynamic> json) => HomeState.fromJson(json);
```

Comment:
you might want to wrap this in a try/catch in case anything goes wrong:
```dart
@override
HomeState fromJson(Map<String, dynamic> json) {
  try {
    return HomeState.fromJson(json);
  } catch (_) {
    return null;
  }
}
```

Reply:
Done